lib/pdk.rb

```diff
@@ -7,4 +8,10 @@
 require 'pdk/validate'
 require 'pdk/version'
 
-module PDK; end
+module PDK
+  def self.analytics
+    @analytics = PDK::Analytics.build_client(
+      logger: PDK.logger,
+    )
+  end
```

Comment:
Should this be an `||=` situation so we only instantiate one client?

Also I think for `PDK.logger` we add this method to the `PDK` module in `lib/pdk/logger.rb`, should probably be consistent one way or the other?

Reply:
Good call on `||=`, don't know how I missed that!

The reason I've put the accessor in `lib/pdk.rb` is that I'm trying to arrange things so that if/when it does get extracted the whole whole `lib/analytics` tree can be yoinked as is and need nothing more than a simple `sed` to rename it to something generic. To that end, I've refactored it to pass the logger in to the constructor rather than calling `PDK.logger` directly in the methods. When we refactor the configuration loading next, I'll do something similar so that the analytics constructor takes the loaded configuration data rather than handle it internally itself.

Reply:
Cool, that makes sense!